R/data.table.R

```diff
@@ -1739,22 +1739,48 @@ replace_dot_alias = function(e) {
         GForce = FALSE
       } else {
         # Apply GForce
+        noCall_noVars = function(expr) (!is.call(expr) || length(all.vars(expr, max.names=1L))==0) && !dotN(expr)
+        .gshift_ok = function(q) {
+          q_named = match.call(shift, q)
```

Comment:
I would just overwrite `q` in this local scope for simplicity. `x = match.call(f, x)` is more common to encounter, e.g.

https://github.com/search?q=lang%3AR+%2F%5Ba-zA-Z0-9._%5D%2B%5Cs*%28%3D%7C%3C-%29%5Cs*match%5B.%5Dcall%5B%28%5D%5B%5E%29%5D*%2C+%5Ba-zA-Z0-9._%5D%2B%5B%29%5D%2F&type=code

`q_named` was more useful in the "outer" scope where `q` is re-used so overwriting it is bad.